cfme/tests/middleware/test_middleware_utilization.py

```diff
@@ -0,0 +1,99 @@
+    for chart_id in utilization_obj.charts:
+        chart = getattr(utilization_obj, chart_id)
+        # validate legends
+        validate_legends(chart=chart)
+        # validate data for Daily with 1 Week
+        chart.option.set_by_visible_text(op_interval=Option.IN_DAILY, op_week=Option.WK_1_WEEK)
+        validate_data(chart=chart)
+        # validate data for Daily with 2 Weeks
+        chart.option.set_by_visible_text(op_interval=Option.IN_DAILY, op_week=Option.WK_2_WEEK)
+        validate_data(chart=chart)
+        # validate data for Daily with 3 Weeks
+        chart.option.set_by_visible_text(op_interval=Option.IN_DAILY, op_week=Option.WK_3_WEEK)
+        validate_data(chart=chart)
+        # validate data for Daily with 4 Weeks
+        chart.option.set_by_visible_text(op_interval=Option.IN_DAILY, op_week=Option.WK_4_WEEK)
+        validate_data(chart=chart)
+        # validate data for Hourly
+        chart.option.set_by_visible_text(op_interval=Option.IN_HOURLY)
+        validate_data(chart=chart)
+        # validate data for Most Recent Hour, 10 Minutes
+        chart.option.set_by_visible_text(op_interval=Option.IN_MOST_RECENT_HOUR,
+                                         op_minute=Option.MN_10_MINUTE)
+        validate_data(chart=chart)
+        # validate data for Most Recent Hour, 15 Minutes
+        chart.option.set_by_visible_text(op_interval=Option.IN_MOST_RECENT_HOUR,
+                                         op_minute=Option.MN_15_MINUTE)
+        validate_data(chart=chart)
+        # validate data for Most Recent Hour, 30 Minutes
+        chart.option.set_by_visible_text(op_interval=Option.IN_MOST_RECENT_HOUR,
+                                         op_minute=Option.MN_30_MINUTE)
+        validate_data(chart=chart)
+        # validate data for Most Recent Hour, 45 Minutes
+        chart.option.set_by_visible_text(op_interval=Option.IN_MOST_RECENT_HOUR,
+                                         op_minute=Option.MN_45_MINUTE_MINUTE)
+        validate_data(chart=chart)
+        # validate data for Most Recent Hour, 60 Minutes
+        chart.option.set_by_visible_text(op_interval=Option.IN_MOST_RECENT_HOUR,
+                                         op_minute=Option.MN_60_MINUTE)
+        validate_data(chart=chart)
+
+
+def validate_legends(chart):
+    for legend_id in chart.legends:
+        legend = getattr(chart, legend_id)
+        legend.set_active(active=False)
+        assert not legend.is_active, "Unable to disable legend[{}] on chart[{}]"\
+            .format(legend.name, chart.name)
+        legend.set_active(active=True)
+        assert legend.is_active, "Unable to enable legend[{}] on chart[{}]"\
+            .format(legend.name, chart.name)
+
+
+def validate_data(chart):
+    """Validate chart data"""
+    data_chart = chart.list_data_chart()
+    data_table = chart.list_data_table()
+    data_mgmt = chart.list_data_mgmt()
+    for m_d in data_mgmt:
```

Comment:
what is m_d, c_d, t_d? Please use more verbose variable names where they have meaning.